### DISCUSSION TITLE
Covers for Network Cable

### DISCUSSION THREAD

**Discussion by Artemis-Hound** [cc-tweaked/CC-Tweaked#1723] [Q&A] [answered]
Hi, I've been getting back into CC:Tweaked for 1.18.2 and I've noticed there doesn't seem to be any way to cover the networking cables.

I saw that there was an issue created (#451) for this back in 2021, and the consensus there seemed to be that there was no good system to use at the time. There were then additional issues created in the SwitchCraftCC/sc-goodies repo in 2023 (https://github.com/SwitchCraftCC/sc-goodies/issues/32 and https://github.com/SwitchCraftCC/sc-goodies/issues/53) which point back to #451 to show that covers still are not possible.

Is this still the case? It has been a long while and CB Multipart ([CurseForge](https://www.curseforge.com/minecraft/mc-mods/cb-multipart) | [Maven stuff](https://maven.covers1624.net/service/rest/repository/browse/maven-hosted/codechicken/CBMultipart/)) seems like it would be a good use case considering it is the successor to Forge Multipart. As mentioned before I am playing on 1.18.2, so I am highly doubtful I would be able to benefit currently if the feature was added, but has there been a re-assessment into whether a cover system is possible now?

**Comment by SquidDev** [comment]
This is something I would really like to do. The tricky bit is that there's still no multipart APIs which are reliably up-to-date.

As you mention, on Forge there's CB Multipart. However, development of this is pretty tied to the rest of the ChickenBones ecosystem, meaning there's currently no stable release for 1.20.1 (the latest is 1.19.2).

For Fabric, there's libmultipart. I did have a look at adding support for this in #1511, however development of libmultipart seems stalled right now, and is missing some features we really need.

**Reply by SquidDev** [reply, marked as answer]
I guess it's probably worth mentioning [MorePeripherals](https://github.com/JheffersonMarques/MorePeripherals), which adds a EnderIO-like facade system instead - you can disguise a cable to look like a different block, rather than actually having full multipart support.

I've been meaning to add this to CC:T itself forever, just never really found a UX I'm happy with.

**Reply by Artemis-Hound** [reply]
Thank you for the reply, and the information about MorePeripherals! Looks like the 1.18 version doesn't have the facades I want (it was probably added later), but I'll mark your comment at the answer for future viewers. I hope you're able to find an appropriate framework to implement it natively at some point in future :)
